M2/Macaulay2/d/evaluate.d

```diff
@@ -1656,13 +1662,28 @@ breakFun(a:Code):Expr := (
      when e is Error do e else Expr(Error(dummyPosition,breakMessage,e,false,dummyFrame)));
 setupop(breakS,breakFun);
 
-addTestS := setupvar("addTest", nullE); -- will be overwritten in testing.m2
-testfun(c:Code):Expr := (
-    r := applyEE(
-	getGlobalVariable(addTestS),
-	seq(eval(c), locate(codePosition(c))));
-    when r is Error do r else nullE);
-setupop(TestS, testfun);
+addTestFromFile(e:Expr):Expr := (
```

Comment:
I don't understand why this is better. It's much more complicated than before.

Reply:
The current behavior is to store a test's location in a hash table, which is easy to do at top level.  But with this proposal, tests are becoming functions, and so unless we want all the tests that have been loaded from files to have `testing.m2` as their location, we need to fake it somehow in the interpreter.  This function was my attempt at doing that.  I figured we should try to find out the ending position so that `code` will work.